src/LogicApp/LogicApp/Cmdlets/IntegrationAccount/NewAzureIntegrationAccountAssemblyCommand.cs

```diff
@@ -0,0 +1,176 @@
+            {
+                case ParameterSet.ByInputObjectAndContentLink:
+                case ParameterSet.ByInputObjectAndFileBytes:
+                case ParameterSet.ByInputObjectAndFilePath:
+                {
+                    var parsedResourceId = new ResourceIdentifier(this.ParentObject.Id);
+                    this.ResourceGroupName = parsedResourceId.ResourceGroupName;
+                    this.ParentName = parsedResourceId.ResourceName;
+                    break;
+                }
+                case ParameterSet.ByResourceIdAndContentLink:
+                case ParameterSet.ByResourceIdAndFileBytes:
+                case ParameterSet.ByResourceIdAndFilePath:
+                {
+                    var parsedResourceId = new ResourceIdentifier(this.ParentResourceId);
+                    this.ResourceGroupName = parsedResourceId.ResourceGroupName;
+                    this.ParentName = parsedResourceId.ParentResource.Split('/')[1];
+                    break;
+                }
+            }
+
+            var assemblyDefinition = new AssemblyDefinition
+            {
+                Properties = new AssemblyProperties
+                {
+                    AssemblyName = this.Name,
+                    ContentType = "application/octet-stream",
+                    Metadata = this.Metadata
+                }
+            };
+
+            switch (this.ParameterSetName)
+            {
+                case ParameterSet.ByInputObjectAndContentLink:
+                case ParameterSet.ByResourceIdAndContentLink:
+                case ParameterSet.ByIntegrationAccountAndContentLink:
+                {
+                    assemblyDefinition.Properties.ContentLink = new ContentLink
+                    {
+                        Uri = this.ContentLink
+                    };
+                    assemblyDefinition.Properties.Content = null;
+                    break;
+                }
+                case ParameterSet.ByInputObjectAndFileBytes:
+                case ParameterSet.ByResourceIdAndFileBytes:
+                case ParameterSet.ByIntegrationAccountAndFileBytes:
+                {
+                    assemblyDefinition.Properties.Content = this.AssemblyData;
+                    assemblyDefinition.Properties.ContentLink = null;
+                    break;
+                }
+                case ParameterSet.ByInputObjectAndFilePath:
+                case ParameterSet.ByResourceIdAndFilePath:
+                case ParameterSet.ByIntegrationAccountAndFilePath:
+                {
+                    assemblyDefinition.Properties.Content = CmdletHelper.GetBinaryContentFromFile(this.TryResolvePath(this.AssemblyFilePath));
+                    assemblyDefinition.Properties.ContentLink = null;
+                    break;
+                }
+            }
+
+            if (this.ShouldProcess(this.Name, $"Creating a new Assembly in resource group '{this.ResourceGroupName}' with name '{this.Name}'."))
```

Comment:
Please put any string that a user will see into a Resources.resx file (this does not include help message in the parameter descriptions, just warnings and errors a user might see while running the cmdlets).

Reply:
Will do